model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/PersistentUserSessionProvider.java

```diff
@@ -1015,8 +1015,12 @@ public static UUID createClientSessionUUID(String userSessionId, String clientId
 
     @Override
     public void migrate(String modelVersion) {
-        if (new ModelVersion(modelVersion).equals(new ModelVersion("25.0.0"))) {
-            migrateNonPersistentSessionsToPersistentSessions();
+        // Changed encoding from JBoss Marshalling to ProtoStream.
+        // Unable to read the cached data.
+        if ("25.0.0".equals(modelVersion) || "26.0.0".equals(modelVersion)) {
+            InfinispanConnectionProvider.DISTRIBUTED_REPLICATED_CACHE_NAMES.stream()
+                    .map(cacheName -> session.getProvider(InfinispanConnectionProvider.class).getCache(cacheName))
+                    .forEach(Cache::clear);
```

Comment:
There seems to be a misunderstanding of how this works, and this is most likely to the API I created here. 

This `migrate()` is not called on every version, but only when it is called explicitly. The only place it is called is in `MigrateTo25_0_0.java`, and there it is always called with `25.0.0`. 

To have the behavior of always clearing the caches for the upgrade to KC26, the following would be necessary:
* Create a class `MigrationTo26_0_0.java and register it so it is called
* Then call the `migrate()` method.

In some earlier version in this or another PR we had a setup where we would open the Infinispan configuration with the JBoss Marshaller and then migrate the sessions. If we're not doing this, we wouldn't allow people to upgrade from KC24 to KC26 in one go, which would mean that a smooth migration wouldn't be available for RHBK users as there is no RHBK25. 

If we choose to not support the straight migration, the code in `PersistentUserSessionProvider.migrate()` should change as follows: 
* Update the code in `PersistentUserSessionProvider` to have a NOOP for `25.0.0`
* Clear the caches for `26.0.0` as the current code shows

Still, this clearing of the caches seems to be a bit unrelated to `PersistentUserSessionProvider` as it is clearing several other caches outside of the user sessions. Maybe the InfinispanConnectionProvider() should be doing this kind of work?

Reply:
In the meeting last month, we agreed on not performing a migration since if users want to save their sessions, they should enable persistent sessions in KC25. Are we changing our minds? :)

Reply:
I have to admit I wasn't sure and possibly got confused with the snipped that was starting the second cache manager. If we then still agree on that we don't offer a migration for RHBK, then so be it. 

Then the only bit remains where to call the clearing of caches.